desc/grid.py

```diff
@@ -774,6 +744,14 @@ def _create_nodes(
             node spacing, based on local volume around the node
 
         """
+        if not (L == M or L == 2 * M):
+            warnings.warn(
```

Comment:
do we need this warning? As intermediate steps in a continuation method its pretty common to not have L=M or L=2M

Reply:
The motivation to add this came from a [surface average test](https://github.com/PlasmaControl/DESC/pull/251#discussion_r960129270) not passing on concentric grids where L was not equal to M or 2*M. I think we should include a warning like this somewhere. Or at the very least add it to the docs.

Reply:
is it an issue with L,M or with L_grid, M_grid? I would think as long as L_grid > L, M_grid > M it should be ok? If it is an issue we should think about how that effects the continuation method

Reply:
It's as issue with L, M. I thought we only needed the grid resolution higher than spectral too, which is why this confused me.
For convenience here is the test that had failed.
```
def test_symmetry_1(self):
    def test(grid, err_msg):
        t = grid.nodes[:, 1]
        z = grid.nodes[:, 2] * grid.NFP
        true_avg = 5
        f = (
            true_avg
            + np.cos(t)
            - 0.5 * np.cos(z)
            + 3 * np.cos(t) * np.cos(z)
            - 2 * np.sin(z) * np.sin(t)
        )
        numerical_avg = surface_averages(grid, f)
        np.testing.assert_allclose(numerical_avg, true_avg, err_msg=err_msg)

    L, M, N, NFP, sym = 3, 6, 3, 3, True
    # passes
    test(LinearGrid(L, M, N, NFP, sym), "LinearGrid")
    # passes
    test(QuadratureGrid(L, M, N, NFP), "QuadratureGrid")
    # fails
    test(ConcentricGrid(L, M, N, NFP, sym), "ConcentricGrid")
```
Changing L to = M = 6 let concentric grid pass too.

Reply:
In order for the code to always behave as expected, we need `L_grid=M_grid` (ANSI) or `L_grid=2*M_grid` (fringe). Basically we want to ensure that the outermost surface of the concentric grid always has `2*M_grid+1` nodes. The surface averages will not give the correct results otherwise, but in some cases (equilibrium solve with fixed-iota) that might not be an issue. Since we can't guarantee the code will behave as expected in that case, I wanted to include this warning.

Reply:
ok, still seems kinda weird to me. Shouldn't the outer ring always have 2*M_grid+1 nodes? 

If it is an issue, I'd recommend at least moving the warning inside the surface_average function or something, so it only fires if its actually an issue in the problem being solved

Reply:
OK, think I fixed it by just adding an extra node sometimes to ensure there's always an odd number of nodes on each surface.